server/server/meta/src/main/java/com/alipay/sofa/registry/server/meta/provide/data/NodeOperatingService.java

```diff
@@ -33,34 +33,45 @@
  */
 public class NodeOperatingService {
 
-  @Autowired protected ProvideDataService provideDataService;
+    @Autowired
+    protected ProvideDataService provideDataService;
 
-  public NodeOperatingService() {}
+    protected Tuple<Long, NodeServerOperateInfo> currentVersion = null;
 
-  public NodeOperatingService(ProvideDataService provideDataService) {
-    this.provideDataService = provideDataService;
-  }
-
-  public Tuple<Long, NodeServerOperateInfo> queryOperateInfoAndVersion() {
-    DBResponse<PersistenceData> response =
-        provideDataService.queryProvideData(ValueConstants.NODE_SERVER_OPERATING_DATA_ID);
+    public NodeOperatingService() {
+    }
 
-    if (response.getOperationStatus() == OperationStatus.NOTFOUND) {
-      return null;
+    public NodeOperatingService(ProvideDataService provideDataService) {
+        this.provideDataService = provideDataService;
     }
 
-    String entityData = PersistenceDataBuilder.getEntityData(response.getEntity());
-    NodeServerOperateInfo read = JsonUtils.read(entityData, NodeServerOperateInfo.class);
-    return new Tuple<>(response.getEntity().getVersion(), read);
-  }
+    public Tuple<Long, NodeServerOperateInfo> queryOperateInfoAndVersion() {
+        DBResponse<PersistenceData> response =
+                provideDataService.queryProvideData(ValueConstants.NODE_SERVER_OPERATING_DATA_ID);
+
+        if (response.getOperationStatus() == OperationStatus.NOTFOUND) {
+            return null;
+        }
+
+        String entityData = PersistenceDataBuilder.getEntityData(response.getEntity());
 
-  /**
-   * Setter method for property <tt>provideDataService</tt>.
-   *
-   * @param provideDataService value to be assigned to property provideDataService
-   */
-  @VisibleForTesting
-  public void setProvideDataService(ProvideDataService provideDataService) {
-    this.provideDataService = provideDataService;
-  }
+        if (null != currentVersion) {
+            if (response.getEntity().getVersion() <= currentVersion.o1) {
```

Comment:
this currentVersion may not be concurrency safe.

Reply:
Why worry about concurrency when there is no multi-value write operation?

Reply:
The method queryOperateInfoAndVersion() is a public method, we cannot assume that the queryOperateInfoAndVersion() method will always be called by a single thread. If queryOperateInfoAndVersion() call concurrency, the currentVersion will write concurrency.

Reply:
All updates currentVer from db, Concurrent writes are also not a problem, right?

Reply:
And db config is updated by Optimistic Locking with version